### ISSUE TITLE
Open source project using a registered trademark

### ISSUE THREAD

**Issue by JakeLin** [JakeLin/SwiftWeather#65]
I received an email from [Swift Weather Company](www.swiftweather.com). They said they are the owner of  U.S. Trademark Registration No. 4175355 for the mark SWIFT WEATHER. And the open source project is in direct conflict with their federal trademark rights. They requested me to change the name of this application and refrain from further use of their mark for any purpose thereafter.

I don't earn any money from this project (even one cent😅). I am seeking advice to handle this issue. I have some potential ideas as below:

1. Modify the [README.md](https://github.com/JakeLin/SwiftWeather/blob/master/README.md) file to add some description of `Swift Weather Company`. Some of the open source projects have sponsors in their README.md. We can do something similar but without charging `Swift Weather Company`. That can help `Swift Weather Company` attract more user to use their services. 
2. Rename the project. Because there are some external links to the current project from other GitHub projects, blog posts etc. It may cause some problem with the redirection. If we need to rename the project, there some options:
    * `SwiftWeatherApp`: I am not sure is it same as the current project name `SwiftWeather` because it still has `SwiftWeather` within the name.
    * `SwiftiOSWeather`: It is a little bit weird for me by adding `iOS` in between. But that should be different from the `SWIFT WEATHER` trademark.
    * Any better name?

I prefer the first option but am seeking more advice to create a win-win situation.

**Comment by doedje** [comment]
Hi Jake,

Very unfortunate. But there are bigger problems in the world I guess....

To react on the options you propose:

1) I think (I am not a lawyer, this is in no way intended to be legal advice, etc, etc...) you can only do something like this with the company's explicit consent. You could do them a proposal, send them a example text to be included with your project. And ask them if they could be okay with that... (I guess they won't, but asking won't harm anybody)

2) Changing the name is the easy solution. I'd like to propose **SwiftyWeather** (didn't check on that one for any trademarks...). You should than move this project to a new github repo `JakeLin/SwiftyWeather`. In the `JakeLin/SwiftWeather` repo you just only put a README.md with a text like: _"SwiftWeather is now SwiftyWeather to avoid a trademark issue with Swift Weather Company."_ They can't complain about that, it's just fair to have you explain the situation to people that follow any old link on the internet and redirect them to either your project under it's new name, or the website of Swift Weather Company.

I hope this helps you decide what to do.

**Comment by JakeLin** [comment]
@doedje thank you very much for your advice, let me see I can do something with the company. **SwiftyWeather** is a good name too 👍 I like it.

**Comment by JakeLin** [comment]
I got back to Swift Weather company and asked them to put their descriptions in the README.md to create a win-win situation. But they refused and asked me to change the name. And for whatever reason, they don't like **SwiftyWeather** and said *It's almost identical to my client's mark.* I have no idea how this **SwiftyWeather** conflict with *Swift Weather* trademark. I am very disappointed when I try to do some good to the others and <del>got f*cked up by someone.</del> (sorry 😞). As a developer, I don't have a layer and don't want to do some like http://www.zdnet.com/article/disgruntled-developer-breaks-thousands-of-javascript-node-js-apps/ Offcause I can't and I shouldn't 😞

Any ideas for the name if I have to change it. Thanks

**Comment by kywix** [comment]
they are just using the right to enforce, a silly attitude that lawyers are accustomed doing since they get paid for.

Here my suggestions:

- WeatherInSwift
- Weather for Swift
- Weather made Swifty
- SwiftTempo (Weather Swift in Italian)

**Comment by pjebs** [comment]
I've got a law degree and studied IP law (albeit Australian IP law). This is my take on the matter based on my memory of what I learn 8+ years ago.

- This is a trademark issue.
- The company has a US trademark only. You are entitled to use it outside of US no questions asked, irrespective of what their lawyers tell you, unless they filed for the trademark globally.
- Finally, you repo was called "Swift Weather". You aren't using the term as a trademark. It is a generic description of what your project is: A weather app written in Swift language.
- Since Apple released the Swift programming language (https://trademarks.justia.com/864/25/swift-86425725.htm),
Swift Weather company's trademark may actually be now unenforceable against weather apps written in swift. *If* it is enforceable, it may very soon 5+ years become unenforceable as more and more people program in the Swift language.

**Comment by JakeLin** [comment]
@pjebs thanks for your useful information. I thought about the trademark is in the US only. And I can register my own track mark in Australia using my ABN too. But I decided to move on and try to build more tech stuff instead of spending more time on something I am not interested into (legal stuff😞). I didn't go CocoaHeads after I had the second one as I need to nurse them at night. I probably will go later once I have more time. Please tap on my shoulder if I have a chance to see you. I'd like to say "thank you" in person 😁.

**Comment by pjebs** [comment]
It is almost certainly a US trademark only, since they didn't file a "Madrid Protocol application".

A Madrid Protocol application makes it more cost-effective to register a trademark for numerous countries in one go. If they didn't do that, they almost certainly don't have a trademark elsewhere.

That means that outside of the US, you have zero trademark issues.

**Comment by pjebs** [comment]
That would be funny. Register the trademark in Australia and then send their lawyers a letter saying that they are infringing on your trademark!

**Comment by JakeLin** [comment]
I offered that to them before for free. And they refused.
 > add in a sweetener where you link to their site on your Readme.md

I just describe the story in README.md. And they (at least the layer) are happy once I changed the name. Anyway, I am not going to make money like this. I do stuff for fun. But off-cause, I am happy to get money if I am not doing something wrong 😅.

**Comment by pjebs** [comment]
They refused because they knew you didn't know your rights or you didn't highlight to them that you knew their lack or remedies in practice.

**Comment by JakeLin** [comment]
@pjebs Yes, I kind of knew they own the trademark in the US only. But I decided to move on to do more fun tech stuff as I mentioned above. Thanks for your advice and ideas.
